File: app/src/main/java/com/faendir/lightning_launcher/multitool/settings/DefaultPlayerPreference.java
package com.faendir.lightning_launcher.multitool.settings;

import android.content.Context;
import android.preference.ListPreference;
import android.util.AttributeSet;

/**
 * @author F43nd1r
 * @since 08.11.2016
 */

public class DefaultPlayerPreference extends ListPreference implements SummaryPreference, HasPlayerEntries {

    public DefaultPlayerPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
        discoverPlayers(context.getPackageManager());
    }

    @Override
    public CharSequence getSummaryText() {
        return getEntry();
    }
}

File: app/src/main/java/com/faendir/lightning_launcher/multitool/settings/DefaultPlayerPreference.kt
package com.faendir.lightning_launcher.multitool.settings

import android.content.Context
import android.util.AttributeSet
import androidx.preference.ListPreference

/**
 * @author F43nd1r
 * @since 08.11.2016
 */

class DefaultPlayerPreference(context: Context, attrs: AttributeSet) : ListPreference(context, attrs), SummaryPreference, HasPlayerEntries {

    override val summaryText: CharSequence
        get() = entry

    init {
        discoverPlayers(context.packageManager)
    }
}
